api/v1alpha8/types.go

```diff
@@ -301,39 +301,147 @@ type LoadBalancer struct {
 	Tags []string `json:"tags,omitempty"`
 }
 
-// SecurityGroup represents the basic information of the associated
+// SecurityGroupStatus represents the basic information of the associated
 // OpenStack Neutron Security Group.
-type SecurityGroup struct {
-	Name  string              `json:"name"`
-	ID    string              `json:"id"`
-	Rules []SecurityGroupRule `json:"rules"`
+type SecurityGroupStatus struct {
```

Comment:
Why the rename for this? It seems strange to me. Based on the name I would expect this to be a status sub-resource of a SecurityGroup. It also feels a bit weird that the `ControlPlaneSecurityGroup` is of type `SecurityGroupStatus` :raised_eyebrow:

Reply:
We needed to differentiate the Status and the Spec. Her we have the ID, so it's a Status; which is why ControlPlaneSecurityGroup is a status type, because it contains the ID and the rules.

@mdbooth help me here :D we wrote this together I think.

Reply:
This object seems to only be used in as status meanwhile the old object seems to have been used for also defining the default rules. [See the code here]( https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1826/files/685696707236c7ec0987adfbb3ea4c5e119719ca#diff-b443ce0ed208632e1331c31b89e3a4f7bf99dd949c15839f920545d55cd5e96eL79-L116)

Reply:
Why not just `SecurityGroup`? Then the ControlPlaneSecurityGroup would have the type SecurityGroup and you would still have the separate struct for SecurityGroupSpec

Reply:
I think this rename may be a vestige of an intermediate state of this patch when we had `SecurityGroup` both in the spec and the status and we realised they needed to be different. This is no longer the case, so I think you're right and it would be clearer to change it back.

Also we should likely rename `SecurityGroupsSpec` to `ManagedSecurityGroups` and move it out of `types.go` into `openstackcluster_types.go`.

Reply:
ok i'll make that change now

Reply:
I'll rename SecurityGroupsSpec to ManagedSecurityGroups but I don't understand why you want to move it out of types.go.